laceworksdk/api/v2/activities.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+"""
+Lacework Activities API wrapper.
+"""
+
+from laceworksdk.api.search_endpoint import SearchEndpoint
+
+
+class ActivitiesAPI:
+
+    def __init__(self, session):
+        """
+        Initializes the ActivitiesAPI object.
+
+        :param session: An instance of the HttpSession class
+
+        :return ActivitiesAPI object.
+        """
+
+        super().__init__()
+        self._base_path = "Activities"
+
+        self.changed_files = ChangedFilesAPI(session, self._base_path)
+        self.connections = ConnectionsAPI(session, self._base_path)
+        self.dns = DnsAPI(session, self._base_path)
+        self.user_logins = UserLoginsAPI(session, self._base_path)
+
+
+class ChangedFilesAPI(SearchEndpoint):
```

Comment:
this feels like a lot of boiler plate... since you are not really doing anything with the `SearchEndpoint` except to provide a single variable, the resource... why not define inside the SeachEndpoint class

```python
class SearchEndpoint:
...
  # If defined this is the resource used.... some comment here
  RESOURCE = ""
...

  def search(...resource=None,...):
..
     if not resource and self.RESOURCE:
        resource = self.RESOURCE

```

and that way you can simply do hwere:

```python
class ChangedFilesAPI(SearchEndpoint):
   """
   Here is my docstring describing the class.  
  """
  RESOURCE = "ChangedFiles"
```

and then you don't have to provide the `__init__` function and the separate `search` ones, since they don't really provide much additional details here.

you can even skip the `resource` parameter from search if you expect all child classes to implement the `RESOURCE` value

Reply:
🔥 updated the structure of the new classes